edxval/api.py

```diff
@@ -974,6 +974,19 @@ def create_transcripts_xml(video_id, video_el, resource_fs, static_dir):
     if video_transcripts.exists():
         transcripts_el = SubElement(video_el, 'transcripts')
 
+    # Create static directory based on the file system's subdirectory,
+    # falling back to default path in case of an error
+    try:
+        # File system should not start from /draft directory.
+        static_file_dir = combine(resource_fs._sub_dir.split('/')[1], static_dir)  # pylint: disable=protected-access
```

Comment:
@DawoudSheraz it appears that it's possible for `resource_fs` to be a `WrapFS` instead of the expected `SubFS` at times, and `WrapFS` doesn't have the private `_sub_dir` attribute.  We're working on developing a fix now, but wondering if you had anything to add regarding the reason behind the particular combination of paths here (second path segment of the directory joined with the static_dir) and what kind of errors it should fallback on (what does KeyError catch here?)

Reply:
> @DawoudSheraz it appears that it's possible for `resource_fs` to be a `WrapFS` instead of the expected `SubFS` at times, and `WrapFS` doesn't have the private `_sub_dir` attribute. We're working on developing a fix now, but wondering if you had anything to add regarding the reason behind the particular combination of paths here (second path segment of the directory joined with the static_dir) and what kind of errors it should fallback on (what does KeyError catch here?)

Hi. What are the instances in the export where the resource_fs won't be SubFS? As for the join statements, the courses with old key format `org/run` don't have `course` as the root directory for export. It was the course run number. After the split, 2nd element in the list is the course run(which is the root directory for old-key format courses). For new courses, the same expression evaluates to `course`.

Reply:
Thanks @DawoudSheraz !  We're still trying to reproduce the issue, so aren't sure exactly yet. I _think_ it's related to one of the `override_export_fs` functions ([here](https://github.com/edx/edx-platform/blob/72aa58406b26f117a40fb259b1aee0f42e962e95/openedx/core/djangoapps/olx_rest_api/adapters.py#L100-L129) or [here](https://github.com/edx/edx-platform/blob/72aa58406b26f117a40fb259b1aee0f42e962e95/openedx/core/djangoapps/xblock/runtime/serializer.py#L96-L132). These both set a WrapFS filesystem, and are used in some functions that serialize xblocks ([here](https://github.com/edx/edx-platform/blob/72aa58406b26f117a40fb259b1aee0f42e962e95/openedx/core/djangoapps/xblock/runtime/serializer.py#L40) and [here](https://github.com/edx/edx-platform/blob/72aa58406b26f117a40fb259b1aee0f42e962e95/openedx/core/djangoapps/olx_rest_api/block_serializer.py#L83).  Perhaps it's only an issue when running the export olx api?

Do think think there will be a simple method that will support both WrapFS and SubFS (either using only public methods defined on both, or a fallback if `._sub_dir` is not available)?  I'm not familiar with how these virtual filesystems are used, so don't know the practical difference between the two for the purposes used here.

Reply:
@DawoudSheraz I investigated further and discovered that indeed a WrapFS is used for use with the olx_rest_api export method.  I opened https://github.com/edx/edx-val/pull/267 with a fix. :)